Okta.AspNet/OktaMiddlewareExtensions.cs

```diff
@@ -132,6 +132,13 @@ private static Task BeforeRedirectToIdentityProviderAsync(RedirectToIdentityProv
                 }
             }
 
+            // Add sessionToken to provide custom login
+            if (n.ProtocolMessage.RequestType == OpenIdConnectRequestType.Authentication && n.OwinContext.Authentication.AuthenticationResponseChallenge != null)
+            {
+                n.OwinContext.Authentication.AuthenticationResponseChallenge.Properties.Dictionary.TryGetValue("sessionToken", out var sessionToken);
+                n.ProtocolMessage.SetParameter("sessionToken", sessionToken);
```

Comment:
Might be able to skip the null check and do:

```csharp
n.OwinContext.Authentication.AuthenticationResponseChallenge?.Properties?.Dictionary?.TryGetValue...

if (!string.IsNullOrEmpty(sessionToken))
{
  n.ProtocolMessage.SetParameter...
}
```

Reply:
Even if the null check syntax can't be simplified, we should check that session token `!IsNullOrEmpty` before setting it as a parameter, just in case.